account_brand/models/account_invoice.py

```diff
@@ -16,3 +16,10 @@ class AccountInvoice(models.Model):
             'cancel': [('readonly', True)],
         }
     )
+
+    @api.constrains('brand_id', 'company_id', 'type')
+    def _check_brand_requirement(self):
```

Comment:
IMO, it's better to override `_is_brand_required`. 


https://github.com/OCA/brand/blob/e7d0e3c1a48445a3a001b6d1343d4f999923e5c9/brand/models/res_brand_mixin.py#L30-L35